src/main/java/EntryPoint/utils/OTPUtil.java

```diff
@@ -0,0 +1,24 @@
+package EntryPoint.utils;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.stereotype.Component;
+
+import java.security.SecureRandom;
+
+@RequiredArgsConstructor
+@Component
+public class OTPUtil {
+    private final StringRedisTemplate redisTemplate;
+
+    public String generateOTP(String email){
+        SecureRandom secureRandom = new SecureRandom();
+        String otp;
+
+        do {
+            otp = String.format("%04d", secureRandom.nextInt(10000));
+        } while (redisTemplate.hasKey(email));
+
```

Comment:
Pass an extra parameter named ttlSeconds in generateOTP() func.
Please store the OTP in Redis with a TTL (Time to Live).
You can use this : redisTemplate.opsForValue().set(email, otp, ttlSeconds, TimeUnit.SECONDS);